code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm

```diff
@@ -3,6 +3,75 @@
 //	#define DEBUG_HIVELORD_ABILITIES
 //#endif
 
+// ***************************************
+// *********** Recycle
+// ***************************************
+/datum/action/xeno_action/activable/recycle
+	name = "Recycle"
+	action_icon_state = "recycle"
+	desc = "We deconstruct the body of a fellow fallen xenomorph to avoid marines from harvesting our sisters in arms. Gain back a few larva points."
+	use_state_flags = XACT_USE_STAGGERED //can't use while staggered, defender fortified or crest down
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_RECYCLE,
+	)
+	plasma_cost = 150
+	gamemode_flags = ABILITY_DISTRESS
+
+/datum/action/xeno_action/activable/recycle/can_use_ability(atom/target)
+	. = ..()
+	var/mob/living/carbon/xenomorph/hivelord = owner
+	var/mob/living/carbon/xenomorph/victim = target
+	if(!.)
+		return FALSE
+	if(hivelord.do_actions)
+		hivelord.balloon_alert(hivelord, "Busy")
+		return FALSE
```

Comment:
This should already be checked by parent, unless you explicitly give it the override.